dev/validate_bom__test.py

```diff
@@ -735,6 +818,13 @@ def validate_test_requirements(self, test_name, spec, metric_labels):
       raise_and_log_error(
           UnexpectedError('Test "{name}" is missing an "api" spec.'.format(
               name=test_name)))
+
+    if spec['api'] in self.__service_endpoints:
```

Comment:
I think this is premature. You still want to check the requirements are satisfied. I think this comes into play within wait_on_service() ~#578